raven/processes/wps_generic_zonal_stats.py

```diff
@@ -77,9 +81,17 @@ def _handler(self, request, response):
         vectors = ['.gml', '.shp', '.gpkg', '.geojson', '.json']
         vector_file = single_file_check(archive_sniffer(shape_url, working_dir=self.workdir, extensions=vectors))
         rasters = ['.tiff', '.tif']
-        raster_file = single_file_check(archive_sniffer(raster_url, working_dir=self.workdir, extensions=rasters))
 
-        vec_crs, ras_crs = crs_sniffer(vector_file), crs_sniffer(raster_file)
+        if 'raster' in request.inputs:
+            raster_url = request.inputs['raster'][0].file
+            raster_file = single_file_check(archive_sniffer(raster_url, working_dir=self.workdir, extensions=rasters))
+        else:
+            bbox = gis.get_bbox(vector_file)
+            raster_url = 'public__EarthEnv_DEM90_NorthAmerica'
+            raster_file = MemoryFile(gis.get_dem(bbox)).name
```

Comment:
Not sure what you mean here.

Reply:
I don't think the `zonalstats` function can handle a `MemoryFile` using the special `MemoryFile` access methods that `rasterio` uses. As such, we supply the `name` of the file which then opens it in the traditional manner. I need to verify whether this is actually faster than using `tempfile` and then, if so, I need to open an issue with the `python-rasterstats` maintainer (who is also a key maintainer of `rasterio`) to see if the `zonalstats` library leverages `MemoryFile`s. If not, I'll open a ticket and if it's really simple to implement, I'll contribute it as a PR.

Reply:
Not sure I understand why you think zonalstats cannot handle it. It does work, doesn't it ?

Reply:
If you send a `MemoryFile` directly to `zonalstats` you get very strange read/access errors and the fact that the `rasterio` docs don't mention using the `.name` to open these files leads me to believe that `.name` is an artefact. I think I just need to read/test more. I might be completely wrong.

Reply:
zonalstats expects a file name, and a Memory file is an object in memory, hence the errors. But my understanding is that MemoryFile.name is a kind of virtual name for the object in memory. zonalstats opens that name and gets access to the memory object directly.

Reply:
My point is, there is no way this works by random chance.

Reply:
Touché. I've opened an issue to ask whether or not there is a penalty for using `.name` in lieu of the `MemoryFile` methods on the `python-rasterstats` repo issues. Once I have a definite answer, this can be put to rest.